compiler/lib/freevars.ml

```diff
@@ -207,10 +207,45 @@ let free_variables vars in_loop p =
   !freevars
 
 let f p =
+  Code.invariant p;
+  let t = Timer.make () in
+  let bound = Code.Var.ISet.empty () in
+  let visited = BitSet.create' p.free_pc in
+  let free_vars =
+    Code.fold_closures_innermost_first
+      p
+      (fun _name_opt params (pc, args) acc ->
+        let free = ref Var.Set.empty in
+        let using x =
+          if Code.Var.ISet.mem bound x then () else free := Var.Set.add x !free
+        in
+        let rec traverse pc =
+          if not (BitSet.mem visited pc)
+          then (
+            BitSet.set visited pc;
+            let block = Addr.Map.find pc p.blocks in
+            List.iter params ~f:(fun x -> Code.Var.ISet.add bound x);
+            iter_block_bound_vars (fun x -> Code.Var.ISet.add bound x) block;
+            List.iter args ~f:using;
```

Comment:
Shouldn't we move these two lines outside of function `traverse`?
```ocaml
            List.iter params ~f:(fun x -> Code.Var.ISet.add bound x);
            List.iter args ~f:using;
```

Reply:
Indeed, fixed in #1565